### PR TITLE
Computer refactor part 2

### PR DESCRIPTION
**Pull request by Qrox**
#### Summary
SUMMARY: None

#### Purpose of change
Continues #37269. Refactor `class computer` to use JSON (de)serialization, and lift the re-ordering restriction on `enum computer_action` and `enum computer_failure_type`. Also cleaned up other code a bit and fixed a bug with COMPFAIL_DESTROY_DATA.

#### Describe the solution
It now saves and loads computer data with actual json objects instead of a hacked-together string. `computer_action` and `computer_failure_type` are now saved as string ids instead of integers. Legacy data are correctly handled and converted to the new structure.

#### Testing
Loaded saves with the legacy save format and computers were all working correctly. Saved and loaded with the new save format and computers were still working as intended.

#### Additional context
This requires #37269.

### CODE REVIEW

File: src/computer.cpp [hunk 1]
```diff
@@ -74,38 +94,16 @@ void computer::set_access_denied_msg( const std::string &new_msg )
...
-    }
-
-    data << string_replace( access_denied, " ", "_" ) << ' ';
-
-    return data.str();
+    mission_id = id;
 }
 
-void computer::load_data( const std::string &data )
+static computer_action computer_action_from_legacy_enum( const int val );
```
Comment (ZhilkinSerg) [inline src/computer.cpp]:
```suggestion
static computer_action computer_action_from_legacy_enum( int val );
```

File: src/computer.cpp [hunk 2]
```diff
@@ -74,38 +94,16 @@ void computer::set_access_denied_msg( const std::string &new_msg )
...
+static computer_failure_type computer_failure_type_from_legacy_enum( const int val );
```
Comment (ZhilkinSerg) [inline src/computer.cpp]:
```suggestion
static computer_failure_type computer_failure_type_from_legacy_enum( int val );
```

File: src/computer.cpp (hunk 1 above)
Comment (ZhilkinSerg) [reply src/computer.cpp]: To appease clang-tidy

**Comment by Inglonias** [comment]
neat!

**Pull request merged**